### PR TITLE
Edit conditional expressions in the inspector

### LINKED ISSUE #3435: Edit conditional expressions in the inspector pane

### PR DESCRIPTION
**Pull request by ruggi**
Fixes #3435 

This PR adds the ability to inline-edit conditional expressions via the inspector.

https://user-images.githubusercontent.com/1081051/225343429-e9640374-f81e-44d2-840c-2d8f7bc4aaa0.mp4

The `condition` of the `JSXConditionalElement` is set by creating a `JSXAttributeOtherJavascript` with the verbatim string entered in the text input, which then triggers a reparse.

### CODE REVIEW

File: editor/src/third-party/react-error-overlay/components/CodeBlock.tsx [hunk 1]
```diff
@@ -30,7 +30,7 @@ const secondaryPreStyle = {
   backgroundColor: yellowTransparent,
 }
 
-const codeStyle = {
+export const codeStyle = {
```
Comment (balazsbajorics) [inline editor/src/third-party/react-error-overlay/components/CodeBlock.tsx:33]: I think this should not be exported as this is source code of react-error-overlay, and instead, if there's no relevant entry, add one to `UtopiaStyles`

File: editor/src/components/inspector/sections/layout-section/conditional-section.tsx [hunk 2]
```diff
@@ -126,12 +128,18 @@ export const ConditionalSection = React.memo(({ paths }: { paths: ElementPath[]
...
     (store) => conditionExpressionSelector(store, paths),
     'ConditionalSection condition expression',
   )
 
+  const [conditionExpression, setConditionExpression] = React.useState(originalConditionExpression)
+
+  React.useEffect(() => {
+    setConditionExpression(originalConditionExpression)
+  }, [originalConditionExpression])
+
```
Comment (balazsbajorics) [inline editor/src/components/inspector/sections/layout-section/conditional-section.tsx:142]:
```suggestion
  const [conditionExpression, setConditionExpression] = usePropControlledStateV2(originalConditionExpression)
```

I believe usePropControlledStateV2 achieves exactly this behavior, but without an extra re-render penalty, and no useEffect

File: editor/src/third-party/react-error-overlay/components/CodeBlock.tsx (hunk 1 above)
Comment (ruggi) [reply editor/src/third-party/react-error-overlay/components/CodeBlock.tsx]: 👍

File: editor/src/components/inspector/sections/layout-section/conditional-section.tsx (hunk 2 above)
Comment (ruggi) [reply editor/src/components/inspector/sections/layout-section/conditional-section.tsx]: oh that's very nice, thank you for the heads up 👌

Comment (ruggi) [reply editor/src/components/inspector/sections/layout-section/conditional-section.tsx]: https://github.com/concrete-utopia/utopia/pull/3436/commits/d8d4b5ad806be19b40c4f403e1d23bb2d51a466a

**Review by seanparsons — APPROVED** [review]

**Pull request merged**